modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/actions/PreIssueAccessTokenClientCredentialsGrantTestCase.java

```diff
@@ -0,0 +1,373 @@
+    private static final String NEW_SCOPE_2 = "new_test_custom_scope_2";
+    private static final String NEW_SCOPE_3 = "new_test_custom_scope_3";
+    private static final String NEW_SCOPE_4 = "replaced_scope";
+
+    private static final String SCIM2_USERS_API = "/o/scim2/Users";
+    private static final String ACTIONS_API = "/api/server/v1/actions";
+    private static final String APPLICATION_MANAGEMENT_API = "/api/server/v1/applications";
+    private static final String API_RESOURCE_MANAGEMENT_API = "/api/server/v1/api-resources";
+    private static final String MOCK_SERVER_ENDPOINT = "/test/action";
+
+    private SCIM2RestClient scim2RestClient;
+    private String accessToken;
+    private String clientId;
+    private String actionId;
+    private String applicationId;
+    private String domainAPIId;
+    private String userId;
+    private String roleId;
+    private JWTClaimsSet jwtClaims;
+
+    /**
+     * Initializes Test environment and sets up necessary configurations.
+     *
+     * @throws Exception If an error occurs during initialization
+     */
+    @BeforeClass(alwaysRun = true)
+    public void testInit() throws Exception {
+
+        super.init(TestUserMode.TENANT_USER);
+
+        scim2RestClient = new SCIM2RestClient(serverURL, tenantInfo);
+        restClient = new ActionsRestClient(serverURL, tenantInfo);
+        // TODO: Review if ActionsRestClient should be instantiated, or if the superclass initialization is sufficient
+
+        List<String> customScopes = Arrays.asList(CUSTOM_SCOPE_1, CUSTOM_SCOPE_2, CUSTOM_SCOPE_3);
+
+        ApplicationResponseModel application = addApplicationWithGrantType(CLIENT_CREDENTIALS_GRANT_TYPE);
+        applicationId = application.getId();
+        if (!CarbonUtils.isLegacyAuthzRuntimeEnabled()) {
+            authorizeSystemAPIs(applicationId, new ArrayList<>(Arrays.asList(SCIM2_USERS_API, ACTIONS_API,
+                    APPLICATION_MANAGEMENT_API, API_RESOURCE_MANAGEMENT_API)));
+        }
+        domainAPIId = createDomainAPI(EXTERNAL_SERVICE_NAME, EXTERNAL_SERVICE_URI, customScopes);
+        authorizeDomainAPIs(applicationId, domainAPIId, customScopes);
+
+        addUserWithRole(applicationId, customScopes);
+
+        MockServer.createMockServer(MOCK_SERVER_ENDPOINT);
+        actionId = createPreIssueAccessTokenAction();
+
+        accessToken = retrieveAccessToken(application.getId(), customScopes);
+        jwtClaims = extractJwtClaims(accessToken);
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void atEnd() throws Exception {
+
+        deleteAction(PRE_ISSUE_ACCESS_TOKEN_API_PATH, actionId);
+        restClient = null;
+        deleteRole(roleId);
+        deleteApp(applicationId);
+        deleteDomainAPI(domainAPIId);
+        scim2RestClient.deleteUser(userId);
+        scim2RestClient = null;
+        MockServer.shutDownMockServer();
+        accessToken = null;
+        jwtClaims = null;
```

Comment:
what's the reason to set all these instance variables to null. IMO we don't need to do that.

Reply:
The instance variables were set to null to avoid conflicting behavior, as most other test cases use these clients. This approach has been consistently applied in other test cases as well.

Reply:
ok but I don't see any potential problem. Given this practice has been followed for some time it's fine to proceed